opm/output/eclipse/Summary.cpp

```diff
@@ -102,18 +103,51 @@ measure rate_unit< rt::gas >() { return measure::gas_surface_rate; }
 template<> constexpr
 measure rate_unit< Phase::GAS >() { return measure::gas_surface_rate; }
 
-template< rt phase >
+template< rt phase, bool injection = true >
 inline quantity rate( const fn_args& args ) {
     double sum = 0.0;
 
     for( const auto* ecl_well : args.ecl_wells ) {
-        if( args.wells.wells.count( ecl_well->name() ) == 0 ) continue;
-        sum += args.wells.at( ecl_well->name() ).rates.get( phase, 0.0 );
+        const auto& name = ecl_well->name();
+        if( args.wells.wells.count( name ) == 0 ) continue;
+        const auto v = args.wells.at( name ).rates.get( phase, 0.0 );
+        if( ( v > 0 ) == injection )
```

Comment:
Pretty much. These are the cases:

1. Injection is requested and this is an injecting well (rates are positive).
2. Injection is requested and this is a producer.
3. Production is requested and this is a producer.
4. Production is requested and this is an injector.

Which gives us:

1. `v > 0 == true && injection == true` -> sum increases
2. `v > 0 == true && injection != true )` -> no-op
3. `v > 0 == false && injection != true` -> sum increases
4. `v > 0 == false && injection == true` no-op

So it's a no-op when the well type doesn't matche the boolean parameter.